adb.js

```diff
@@ -604,17 +571,25 @@ const ADB = {
             // handle full chunk
             if (chunkSize > 0 && pkgData.length >= chunkSize) {
               let chunkData = pkgData.subarray(0, chunkSize);
-              fileData.set(chunkData, currentPos);
+              let tmpData = new Uint8Array(currentPos + chunkSize);
+              if (fileData) {
+                tmpData.set(fileData, 0);
+              }
+              tmpData.set(chunkData, currentPos);
+              fileData = new Uint8Array(tmpData);
               pkgData = pkgData.subarray(chunkSize, pkgData.length);
               currentPos += chunkSize;
-              remaining -= chunkSize;
               chunkSize = 0;
             }
             // handle partial chunk at the end of socket package
-            if (chunkSize > 0 && pkgData.length < chunkSize) {
-              fileData.set(pkgData, currentPos);
+            if (chunkSize > 0 && 0 < pkgData.length && pkgData.length < chunkSize) {
```

Comment:
Nit: Please flip this around to `pkgData.length > 0` so that the constant is on the right side